src/test/java/com/coumin/woowahancoupons/coupon/controller/CouponRestControllerTest.java

```diff
@@ -29,100 +38,147 @@
-				.amount(0L)
-				.daysAfterIssuance(0)
-				.minOrderPrice(0L)
-				.build())
-			.collect(Collectors.toList());
-		StoreCouponSaveRequestDto requestDto = new StoreCouponSaveRequestDto(storeCouponSaveDtos);
-
-		//When
-		ResultActions resultActions = requestCreateStoreCoupons(storeId, requestDto);
-
-		//TODO rest docs
-		//Then
-		resultActions
-			.andExpect(status().isBadRequest())
-			.andDo(print());
-	}
-
-	@Test
-	@DisplayName("요청이 유효하지 않은 메서드라면 매장의 쿠폰 생성 실패")
-	void createStoreCouponsInvalidRequestMethodTest() throws Exception {
-		//Given
-		long storeId = storeRepository.save(new Store("testStore#1")).getId();
-		List<StoreCouponSaveDto> storeCouponSaveDtos = IntStream.range(0, 5)
-			.mapToObj(i -> StoreCouponSaveDto.builder()
-				.name("name#" + i)
-				.amount(1000L * (i + 1))
-				.daysAfterIssuance(30)
-				.minOrderPrice(1000L * (i + 1))
-				.build())
-			.collect(Collectors.toList());
-		StoreCouponSaveRequestDto requestDto = new StoreCouponSaveRequestDto(storeCouponSaveDtos);
-
-		//When
-		ResultActions resultActions = mvc.perform(put("/api/v1/coupons/{storeId}/issuance", storeId)
-			.contentType(MediaType.APPLICATION_JSON)
-			.content(objectMapper.writeValueAsString(requestDto)))
-			.andDo(print());
-
-		//TODO rest docs
-		//Then
-		resultActions
-			.andExpect(status().isMethodNotAllowed())
-			.andDo(print());
-	}
-
-	private ResultActions requestCreateStoreCoupons(long storeId,
-		StoreCouponSaveRequestDto requestDto)
-		throws Exception {
-		return mvc.perform(post("/api/v1/coupons/{storeId}/issuance", storeId)
-			.contentType(MediaType.APPLICATION_JSON)
-			.content(objectMapper.writeValueAsString(requestDto)))
-			.andDo(print());
-	}
+    @Autowired
+    private MockMvc mvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private StoreRepository storeRepository;
+
+    @Autowired
+    private CouponAdminRepository couponAdminRepository;
+
+    @Autowired
+    private CouponConverter couponConverter;
+
+    @Test
+    @DisplayName("매장의 쿠폰 생성 성공")
+    void createStoreCouponsSuccessTest() throws Exception {
+        //Given
+        long storeId = storeRepository.save(new Store("testStore#1")).getId();
+        List<StoreCouponSaveDto> storeCouponSaveDtos = IntStream.range(0, 3)
+            .mapToObj(i -> StoreCouponSaveDto.builder()
+                .name("name#" + i)
+                .amount(1000L * (i + 1))
+                .daysAfterIssuance(30)
+                .minOrderPrice(1000L * (i + 1))
+                .build())
+            .collect(Collectors.toList());
+        StoreCouponSaveRequestDto requestDto = new StoreCouponSaveRequestDto(storeCouponSaveDtos);
+
+        //When
+        ResultActions resultActions = requestCreateStoreCoupons(storeId, requestDto);
+
+        //TODO rest docs
+        //Then
+        resultActions
+            .andExpect(status().isOk())
+            .andDo(print());
+    }
+
+    @Test
+    @DisplayName("요청이 유효하지 않으면 매장의 쿠폰 생성 실패")
```

Comment:
어떤 요청이 유효한지 않으면 볼때 더 도움이 될것 같습니다

Reply:
이런 제가 작성한 테스트 코드네요~! 상원님 요렇게 수정 부탁드려도 될까요??
```suggestion
    @DisplayName("매장의 쿠폰 생성 실패 - 생성 수량 초과 및 쿠폰의 값들 0")
```